From 164303b56d5e0f3d9dcac1c6807b6ae7f99efc78 Mon Sep 17 00:00:00 2001
From: Christophe Roullier <christophe.roullier@foss.st.com>
Date: Wed, 15 Nov 2023 18:15:02 +0100
Subject: [PATCH] arm64: dts: st: add sleep pins for Ethernets in
 stm32mp257f-ev1

Add analog gpio pin configuration ("sleep") to manage power mode on
stm32mp25.

Change-Id: I99e08aad282f97f33a81fca06bbd2194c951ee3b
Signed-off-by: Christophe Roullier <christophe.roullier@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -201,7 +201,8 @@
 &eth1 {
 	status = "disabled";
 	pinctrl-0 = <&eth1_mdio_pins_a>;
-	pinctrl-names = "default";
+	pinctrl-1 = <&eth1_mdio_sleep_pins_a>;
+	pinctrl-names = "default", "sleep";
 	phy-mode = "rgmii";
 	max-speed = <1000>;
 	st,eth-clk-sel;
@@ -238,7 +239,8 @@
 &eth2 {
 	status = "okay";
 	pinctrl-0 = <&eth2_rgmii_pins_a>;
-	pinctrl-names = "default";
+	pinctrl-1 = <&eth2_rgmii_sleep_pins_a>;
+	pinctrl-names = "default", "sleep";
 	phy-mode = "rgmii-id";
 	max-speed = <1000>;
 	phy-handle = <&phy1_eth2>;
@@ -620,7 +622,8 @@
 &switch0 {
 	status = "disabled";
 	pinctrl-0 = <&eth1_rgmii_pins_a>, <&eth3_rgmii_pins_a>;
-	pinctrl-names = "default";
+	pinctrl-1 = <&eth1_rgmii_sleep_pins_a>, <&eth3_rgmii_sleep_pins_a>;
+	pinctrl-names = "default", "sleep";
 	phy-mode = "rgmii";
 	st,ethsw-internal-125;
 };
